js/output/shared/loop-protect.js

```diff
@@ -116,9 +120,18 @@ window.LoopProtector.prototype = {
         }
     },
 
-    _KAInfiniteLoopSetTimeout: function (timeout) {
-        this.timeout = timeout;
-        this.branchStartTime = 0;
+    _KAInfiniteLoopSetTimeout: function (timeout, protectorKey) {
+        // Make sure that the key we've been provided matches the one on this
+        // LoopProtector
+        if (protectorKey === this.protectorKey) {
+            this.timeout = timeout;
+            this.branchStartTime = 0;
+        } else {
+            // Let the user know that what they're trying to do is a bad idea
+            throw {
+                html: "Messing with the loop protector is dangerous!"
```

Comment:
I think this might conflict with changes from https://github.com/Khan/live-editor/commit/57130931aa8ded361a081b7cfc803c3bf4d317e0.  Could you rebase and then run the async tests to see if there's an issue?  Also, this message should be wrapped in call to `i18n._`, e.g. `i18n._("Messing with the loop protector is dangerous!")`.